packages/flutter_tools/lib/src/reporting/events.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of reporting;
+
+class UsageEvent {
+  UsageEvent(this.category, this.parameter);
+
+  final String category;
+  final String parameter;
+
+  void send() {
+    flutterUsage.sendEvent(category, parameter);
+  }
+}
+
+class HotEvent extends UsageEvent {
```

Comment:
I think the contextual comments you left in the runner might also make sense as doc comments on this class too

Reply:
Included those comments here, and added comments in various other places as well.